### PR TITLE
[CCXDEV-12259] New publisher for archive sync

### PR DESCRIPTION
**Pull request by Jakub007d**
# Description

New publisher for achive sync

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Unit tests (no changes in the code)

## Testing steps

Tested localy

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: ccx_messaging/publishers/idp_publisher.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    """Publisher for interanal data pipeline."""
+
+    def publish(self, input_msg: Dict, report: str) -> None:
+        """Publish response as Kafka message to outgoing topic."""
+        output_msg = {}
+        try:
+            output_msg = json.loads(report)
+        except (TypeError, json.decoder.JSONDecodeError) as err:
+            raise CCXMessagingError("Could not parse report; report is not in JSON format") from err
+        output_msg.pop("reports", None)
```
Comment (juandspy) [inline ccx_messaging/publishers/idp_publisher.py:20]: why?

Comment (Jakub007d) [reply ccx_messaging/publishers/idp_publisher.py:20]: Just a check if the report was loaded propperly i can remove it from there if you think is not needed

Comment (Jakub007d) [reply ccx_messaging/publishers/idp_publisher.py:15]: Ok i removed it from there

**Review by JiriPapousek — APPROVED** [review]

File: ccx_messaging/publishers/idp_publisher.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import json
+from typing import Dict
+import logging
+from ccx_messaging.publishers.kafka_publisher import KafkaPublisher
+
+LOG = logging.getLogger(__name__)
+
+
+class IDPPublisher(KafkaPublisher):
```
Comment (joselsegura) [inline ccx_messaging/publishers/idp_publisher.py:26]: I would like to find a different name for the file and the class. In the Internal Data Pipeline we can have several publishers, so probable this name is misleading.

Maybe something related to "SyncedArchivePublisher" for example could be better IMHO

File: test/publishers/idp_publisher_test.py
Comment (joselsegura) [inline test/publishers/idp_publisher_test.py:1]: Same naming problem as stated for the the publisher

**Review by joselsegura — CHANGES_REQUESTED** [review]

**Review by joselsegura — APPROVED** [review]

**Pull request merged**